app/src/main/java/com/example/ithaca_transit_android_v2/NetworkingAdapters.kt

```diff
@@ -44,4 +46,128 @@ class LocationAdapter {
         }
         return DataLocation(final)
     }
+}
+
+@JsonClass(generateAdapter = true)
+class RouteAdapter {
+    class DataRoute(
+        val data: DataRouteOptions
+    )
+
+    @JsonClass(generateAdapter = true)
+    class DataRouteOptions(
+        val fromStop: List<JsonRoute>,
+        val walking: List<JsonRoute>,
+        val boardingSoon: List<JsonRoute>
+    )
+
+    @JsonClass(generateAdapter = true)
+    class JsonRoute(
+        @Json(name = "directions")
+        val directions: List<JsonDirection>,
+        @Json(name = "startCoords")
+        val startLocation: JsonLocation,
+        @Json(name = "endCoords")
+        val endLocation: JsonLocation,
+        // TODO: need to calculate isWalkingOnly
+        @Json(name = "arrivalTime")
+        val arrival: String, // e.g. 2019-11-06T17:07:20Z
+        @Json(name = "departureTime")
+        val depart: String
+        // TODO: need to calculate BoardInMin
+    )
+
+    @JsonClass(generateAdapter = true)
+    class JsonDirection(
+        @Json(name = "path")
+        val listOfCoordinates: List<Coordinate>,
+        @Json(name = "startTime")
+        val startTime: String,
+        @Json(name = "endTime")
+        val endTime: String,
+        @Json(name = "startLocation")
+        val startLocation: JsonLocation,
+        @Json(name = "endLocation")
+        val endLocation: JsonLocation,
+        @Json(name = "stops")
+        val busStops: List<JsonLocation>,
+        @Json(name = "routeNumber")
+        val busNumber: Int
```

Comment:
We may need to make a few of these values nullable because we may have to walk. In that case, backend will pass us a null busNumber.

We might also want to add a type variable that takes in an enum between "depart" and "walk".

Reply:
i thought that we decided to set "invalid" `busNumber`s to -1 or 0 to avoid nulls? but you are right in that we need to add an `ActionType` for this model

Reply:
Yeah if this represents a single route in a trip than we need an actionType. I think we should invalid the busNumbers to remove the nulls

Reply:
This `ActionType` enum already exists:
<img width="337" alt="Screen Shot 2019-11-17 at 12 46 44 PM" src="https://user-images.githubusercontent.com/36868927/69011440-64ab7c00-0938-11ea-9273-2bc1e4508b16.png">

The types are as follows:
<img width="421" alt="Screen Shot 2019-11-17 at 12 47 31 PM" src="https://user-images.githubusercontent.com/36868927/69011452-71c86b00-0938-11ea-8c31-51b8b0034b31.png">